File: cs/LeetCode/LeetCode/Array/_BestTimeToBuyAndSellStock.cs
namespace LeetCode.Array
{
    internal class _BestTimeToBuyAndSellStock
    {
        public int MaxProfit(int[] prices)
        {
            int profit = 0;
            int buy = int.MaxValue;

            foreach (int price in prices)
            {
                buy = Math.Min(buy, price);
                profit = Math.Max(profit, price - buy);
            }

            return profit;
        }
    }
}

File: cpp/LeetCode/LeetCode/BestTimeToBuyAndSellStock.cpp
#include<vector>
#include<math.h>

using namespace std;

class BestTimeToBuyAndSellStock {
public:
    int maxProfit(vector<int>& prices) {
        int buy = INT_MAX;
        int profit = 0;

        for (int price : prices) {
            buy = min(buy, price);
            profit = max(profit, price - buy);
        }

        return profit;
    }
};